modules/ROOT/pages/core-operations.adoc

```diff
@@ -300,51 +300,74 @@ Using the append and prepend methods may invalidate an existing JSON document.
 
-The semantics of the _append_ and _prepend_ operations are similar to those of the _upsert_ family of operations, except that they accept the fragment to append as their value, rather than the entire document.
+The semantics of the _append_ and _prepend_ operations are similar to those of the _upsert_ family of operations, except that they accept the document whose content can be appended to the existing one, rather than the entire document.
 These functions may be used to add efficiency for custom binary data structures (such as logs), as they avoid transferring the contents of the entire document for each operation.
 Consider the following versions (which are equivalent)
 
 .Append using get() and replace() (slow)
-[source,python]
+[source,java]
 ----
-# Store the document
-cb.upsert('binary_doc', '\x01', format=couchbase.FMT_BYTES)
+// Create buffer out of a string
+ByteBuf toWrite = Unpooled.copiedBuffer("Hello", CharsetUtil.UTF_8);
+
+// Store the document
+bucket.upsert(BinaryDocument.create("binaryDoc", toWrite));
+
+// Get the document
+BinaryDocument read1 = bucket.get("binaryDoc", BinaryDocument.class);
+
+// Create a replacement document
+ByteBuf toAppend = Unpooled.copiedBuffer("World", CharsetUtil.UTF_8);
+byteBuf toReplace = read1.content().writeBytes(toAppend);
+
+// Replace the existing document
+bucket.replace("binaryDoc",BinaryDocument.create("binaryDoc", toReplace));
 
-while True:
-    # Retrieve the entire document
-    rv = cb.get('binary_doc')
-    value = rv.value + '\x02'
-    try:
-        # Upload the entire document
-        cb.replace('binary_doc', value, format=couchbase.FMT_BYTES)
-        break
-    except couchbase.exceptions.KeyExistsError:
-        continue
+// Read it back
+BinaryDocument read2 = bucket.get("binaryDoc", BinaryDocument.class);
 
-print repr(cb.get('binary_doc').value)
+// Print it
+System.out.println(read2.content().toString(CharsetUtil.UTF_8)); //Hello World
+
+// Free the resources
+ReferenceCountUtil.release(read1.content());
+ReferenceCountUtil.release(read2.content());
 ----
 
 .Append using append() (fast)
-[source,python]
+[source,java]
 ----
-# Store the document
-cb.upsert('binary_doc', '\x01', format=couchbase.FMT_BYTES)
+// Create buffer out of a string
+ByteBuf toWrite = Unpooled.copiedBuffer("Hello", CharsetUtil.UTF_8);
+
+// Store the document
+bucket.upsert(BinaryDocument.create("binaryDoc", toWrite));
+
+// Create a buffer to be appended
+ByteBuf toAppend = Unpooled.copiedBuffer("World", CharsetUtil.UTF_8);
+
+// Append a document
+bucket.append(BinaryDocument.create("binaryDoc", toAppend))
+
+// Get the document
+BinaryDocument read = bucket.get("binaryDoc", BinaryDocument.class);
 
-# Append a fragment
-cb.append('binary_doc', '\x02', format=couchbase.FMT_BYTES)
+// Print it
+System.out.println(read.content().toString(CharsetUtil.UTF_8)); //Hello World
```

Comment:
I doubt it'll be "Hello World" as there's no space in the appended data.  Please do check that all examples do what they say they'll do.

Reply:
Apart from missing `;`, all lines got compiled successfully. Same is corrected now.
Further, i have removed space between `Hello` and `World`. I hope its fine.